Users/Glob_Patterns.rst

```diff
@@ -192,13 +182,41 @@ brackets. Parentheses that have no match are ignored as well as
 
 ::
 
-    >>> fnmatch("abc", "a**c")
-    True
     >>> fnmatch("abbc", "a**c")
     True
     >>> fnmatch("a/c", "a**c")
     True
-    >>> fnmatch("a?c", "a**c")
-    True
-    >>> fnmatch("ac", "a**c")
-    True
+
+Using Glob Patterns
+-------------------
+
+Suppose, You want `SpaceConsistencyBear` to perform an analysis on a file
+`first.c`. You should use the command:
+::
+
+    coala --files=src/\first.c --bears=SpaceConsistencyBear --save
+
+.. note::
+
+    If you don't know the fuctions of a bear or how to perform the analysis
+    with a bear, You should go through `Tutorial
+    <http://coala.readthedocs.io/en/latest/Users/Tutorial.html#ignoring
+    -issues>`_ first.
+
+Now, if you want all the `.c` files in a specific directory to be analysed,
+you can take help of Glob Patterns.
+::
+
+    coala --files=src/\*.c --bears=SpaceConsistencyBear --save
+
+Here, `*.c` matches all `.c` files in the specified directory.
+Going further, if you want all `.c` as well as `.java` files to be analysed:
+::
+
+    coala --files='src/\*.(java|c)' --bears=SpaceConsistencyBear --save
+
+If you want your `files` argument to match all directories and subdirectories,
+you can use `**` Glob Pattern for that.
+::
+
+    files = **/*.c
```

Comment:
Could you add a note here saying that only `.c` files at least one subdirectory will be matched? This issue has been brought up a few times by others - the basic idea is that since we have a `/` in the glob expression, any matching file will need to be at least one subdirectory below (to have the `/`). So no `.c` files in the root directory will be matched.

This is not obvious, but kinda logical. So I think this would be good mention in the docs.